lib/python/rose/host_select.py

```diff
@@ -132,6 +133,38 @@ def __init__(self, event_handler=None, popen=None):
             popen = RosePopener(event_handler=event_handler)
         self.popen = popen
         self.scorers = {}
+        self.local_host_strs = None
+
+    def get_local_host_strs(self):
+        """Return a list of names associated with the current host."""
+        if self.local_host_strs is None:
+            self.local_host_strs = []
+            for item in gethostname, getfqdn, "localhost":
```

Comment:
I didn't know we could do this with a `for` statement!

I think it's better to build up a list of names, then go into the `name not in...` code. That avoids having the `callable` test (and the `if name` one later).